packages/guides/src/ReferenceResolvers/SnakeCaseAnchorReducer.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\ReferenceResolvers;
+
+use function preg_replace;
+use function strtolower;
+use function trim;
+
+class SnakeCaseAnchorReducer implements AnchorReducer
+{
+    public function reduceAnchor(string $rawAnchor): string
+    {
+        $cleanedAnchor = strtolower($rawAnchor);
+
+        // Remove unwanted characters and convert to lowercase
+        $cleanedAnchor = preg_replace('/[^a-z0-9]+/i', '-', $cleanedAnchor);
+
+        // Remove leading and trailing underscores
+        $cleanedAnchor = trim($cleanedAnchor ?? '', '-');
+
+        // Remove consecutive underscores
+        $cleanedAnchor = preg_replace('/-+/', '-', $cleanedAnchor);
+
+        return $cleanedAnchor ?? '';
```

Comment:
I think we can replace this with symfony/string: https://symfony.com/doc/current/components/string.html#slugger

Reply:
Good idea, done